### ISSUE TITLE
Mailing Issues

### ISSUE THREAD

**Issue by seppeel** [Cacti/plugin_routerconfigs#42]
i think it makes sense to create a new issue regarding to mailing problems (low priority :) )
Using routerconfigs 1.3.1 and cacti 1.1.36 (-> PHP Errors are gone in cacti.log)

- A mail is still (or again?) sent one time per hour with 0/0 information (Network Device Configuration Backups - Reattempt)
- The mail is sent from <username>@<hostname> ignoring cacti settings

And by the way the backup is running fine again with 1.3.1 👍 

Edit:

cacti.log
```
27-02-2018 11:00:11 - MAILER Mail Successfully Sent to '<mailaddress>', Subject: 'Network Device Configuration Backups - Reattempt'
--
27-02-2018 11:00:11 - SYSTEM SYSLOG STATS:Time:0.22 Deletes:0 Incoming:0 Removes:0 XFers:9 Alerts:0 Alarms:0 Reports:0
27-02-2018 11:00:10 - RCONFIG STATS: Backup now running since 2018-02-27 11:00:10
27-02-2018 11:00:10 - RCONFIG DEBUG: Executing '/usr/bin/php' with arguments ' -q /var/www/cacti/plugins/routerconfigs/router-download.php --retry'
```

With "--retry"?

**Comment by netniV** [comment]
The email settings are in the Settings -> RouterConfigs tab.  I had heard the 0/0 had resurfaced... I will check why.

Other than midnight, every hour has --retry to download any configuration that failed previously.

**Comment by seppeel** [comment]
Ah ok.
The from address defined in Settings -> Routerconfigs tab is ignored, too :)

**Comment by netniV** [comment]
Hmm, I'll check on that.

**Comment by netniV** [comment]
I did find a problem with that.  I've patched it in my dev repo.  But i'm looking through a few more things before I publish the other changes i've made.

**Comment by netniV** [comment]
Oh and I even added an option to set the from "Name" too.

**Comment by netniV** [comment]
The new update should fix it.

**Comment by seppeel** [comment]
i check when PR #43 is merged and i can pull it, no time for manual updates right now :(

**Comment by netniV** [comment]
No worries, just waiting on @cigamit  for that unfortunately and I know he's quite busy at the moment with other things.

**Comment by netniV** [comment]
This has now been merged 👍

**Comment by cigamit** [comment]
Please close if resolved.

**Comment by seppeel** [comment]
The 0/0 mails are gone, but the the mail are still sent from <username>@<hostname>.

**Comment by netniV** [comment]
Check the Settings -> RouterConfigs tab, make sure you have defined your name/email

**Comment by seppeel** [comment]
Yes, i did. its configured with hostname@domain.de
and the from name is hostname

But the mails are still sent from username@hostname

Github removes my > and < 

edited the post.

**Comment by netniV** [comment]
Use a single backtick around things with `<>` in them and your browser stops stealing them from you ;-)

It's actually not GitHub doing it, it's because they are treated as HTML.

**Comment by netniV** [comment]
I've found another bug in the base cacti code with the from name were that would be wiped out, but the from email address is still used.

If you run the following, do you see your email address details:
```sql
select * from settings where name like 'routerconfigs%'
```
They should be in the _from and _name fields.

**Comment by netniV** [comment]
It may be worth, since you are on the cutting edge, using the following command:
```
git remote set-url origin https://github.com/netniv/plugin_routerconfigs
git pull
```
This will migrate you to my repo with the latest changes.  Note, that this doesn't fix the email name issue as that's still to be merged in the base cacti code.

**Comment by seppeel** [comment]
Yes.

```
MariaDB [cacti]> select * from settings where name like 'routerconfigs%';
+----------------------------+---------------------------------+
| name                       | value                           |
+----------------------------+---------------------------------+
| routerconfigs_backup_path  | /routerconfigs/                 |
| routerconfigs_debug_buffer |                                 |
| routerconfigs_email        | <my_mailaddress>  |
| routerconfigs_from         | <hostname>@<domain.de>     |
| routerconfigs_hour         | 0                               |
| routerconfigs_name         | <hostname>                       |
| routerconfigs_retention    | 30                              |
| routerconfigs_retry        | 0                               |
| routerconfigs_tftpserver   | <ip>                 |
+----------------------------+---------------------------------+
9 rows in set (0.00 sec)
```

It's stored correctly. 
Today im done with this, as im very busy with other stuff, maybe tomorrow i move to your repo.
thanks!

**Comment by netniV** [comment]
OK, no worries.  Question, I noticed you have `<>` in _email, _from and _name.  I'm assuming there are none of those in the actual output?  If not, you may want to try applying https://github.com/Cacti/cacti/pull/1445 and see if that resolves things.  

Oh one final thing, what mailer method are you using? SMTP or Sendmail?

**Comment by seppeel** [comment]
Yes, none of the stored settings is in the actual output. I could try to check it with the cacti patch tomorrow (i hope)
Im using sendmail :)

**Comment by seppeel** [comment]
My colleague had a good idea:
Make a third row in the mail:
- XX devices are disabled

**Comment by netniV** [comment]
Create a separate issue for that and I'll look at it 👍

**Comment by seppeel** [comment]
OK, i now switched to your repo and applied the cacti patch.
 -> This did not change anything. But i sent a testmail from Settings/Mail page and its wrong, too.

So this is clearly no routerconfigs issue.
- Its a cacti issue
- sendmail configured wrong

I now have time to investigate that.

**Comment by seppeel** [comment]
was wrong configured sendmail -.-
Closed.

**Comment by netniV** [comment]
I had a feeling it was.  I've tested the send (using SMTP) quite a bit, so I knew I'd narrowed down the last "From" issue using that.  One more day of running and I'll publish 1.3.4
